### ISSUE TITLE
Math font recommendations sometimes include non-math fonts

### ISSUE THREAD

**Issue by Enivex** [typst/typst#2732]
### Description

If I write `#set math.equation: set text(font: )` then recommendations correctly show only math fonts

![image](https://github.com/typst/webapp-issues/assets/2191754/e765f531-de9b-44cf-97ac-28370c7d6990)

However, if I input "" (which is natural if I want to input a font name, then the list suddenly includes all fonts

![image](https://github.com/typst/webapp-issues/assets/2191754/ca8d519c-d05f-42ff-9e3c-ddbdf78a7455)

This is problematic, because non-math fonts simply don't function properly in math. This has come up several times on the Discord.

**Comment by laurmaedje** [comment]
Checking for a math table necessitates downloading/loading the font, which is too expensive.

Detecting the situation in a more robust way is surely possible, the person who wrote this (I think it was me), just didn't bother enough.

**Comment by Enivex** [comment]
> Checking for a math table necessitates downloading/loading the font, which is too expensive.

In the web app this information could be cached server side could it not?

**Comment by laurmaedje** [comment]
Yes, I forgot to add that part. The web app uses has a font index for these kinds of things, but math information is not currently part of it. Changing the font index while keeping compatibility with old compilers is generally a bit tricky, but in this case would actually be pretty easy. Whether it's a math font could be added to the font flags:

https://github.com/typst/typst/blob/df89a0e85b80844ef56a6fa98af01eaaf7553da8/crates/typst-library/src/text/font/book.rs#L192
